internal/provider/connectivity_rule_resource.go

```diff
@@ -0,0 +1,354 @@
+		return
+	}
+	if err := client.AwaitAsyncOperation(ctx, r.client, svcResp.AsyncOperation); err != nil {
+		resp.Diagnostics.AddError("Failed to create Connectivity Rule", err.Error())
+		return
+	}
+
+	connectivityRule, err := r.client.CloudService().GetConnectivityRule(ctx, &cloudservicev1.GetConnectivityRuleRequest{
+		ConnectivityRuleId: svcResp.ConnectivityRuleId,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get Connectivity Rule after creation", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(updateConnectivityRuleModelFromSpec(&plan, connectivityRule.ConnectivityRule)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+
+}
+
+func (r *connectivityRuleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError(
+		"Update Not Supported",
+		"Connectivity rules cannot be updated. To modify a connectivity rule, it must be destroyed and recreated.",
+	)
+}
+
+func (r *connectivityRuleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state connectivityRuleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connectivityRule, err := r.client.CloudService().GetConnectivityRule(ctx, &cloudservicev1.GetConnectivityRuleRequest{
+		ConnectivityRuleId: state.ID.ValueString(),
+	})
+
+	if err != nil {
+		switch status.Code(err) {
+		case codes.NotFound:
+			tflog.Warn(ctx, "Connectivity Rule Resource not found, removing from state", map[string]interface{}{
+				"id": state.ID.ValueString(),
+			})
+
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError("Failed to get Connectivity Rule", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(updateConnectivityRuleModelFromSpec(&state, connectivityRule.ConnectivityRule)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (r *connectivityRuleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state connectivityRuleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	deleteTimeout, diags := state.Timeouts.Delete(ctx, defaultDeleteTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	currentConnectivityRule, err := r.client.CloudService().GetConnectivityRule(ctx, &cloudservicev1.GetConnectivityRuleRequest{
+		ConnectivityRuleId: state.ID.ValueString(),
+	})
+	if err != nil {
+		switch status.Code(err) {
+		case codes.NotFound:
+			tflog.Warn(ctx, "Connectivity Rule Resource not found, removing from state", map[string]interface{}{
+				"id": state.ID.ValueString(),
+			})
+
+			resp.State.RemoveResource(ctx)
```

Comment:
You don't need to call RemoveResource on delete: https://developer.hashicorp.com/terraform/plugin/framework/resources/delete#recommendations
```suggestion
```